adapter.go

```diff
@@ -247,15 +247,20 @@ func NewAdapterByDB(db *gorm.DB) (*Adapter, error) {
 	return NewAdapterByDBUseTableName(db, "", defaultTableName)
 }
 
-func NewAdapterByDBWithCustomTable(db *gorm.DB, t interface{}) (*Adapter, error) {
+func NewAdapterByDBWithCustomTable(db *gorm.DB, t interface{}, tableName ...string) (*Adapter, error) {
 	ctx := db.Statement.Context
 	if ctx == nil {
 		ctx = context.Background()
 	}
 
 	ctx = context.WithValue(ctx, customTableKey{}, t)
 
-	return NewAdapterByDBUseTableName(db.WithContext(ctx), "", defaultTableName)
+	curTableName := defaultTableName
+	if tableName != nil {
+		curTableName = tableName[0]
+	}
```

Comment:
What about `NewAdapterByDBWithCustomTable(db, t, []string{}...)`
It will panic, I suggest using `len(tableName) > 0`